libparsec/crates/client/src/certif/store.rs

```diff
@@ -655,6 +659,13 @@ macro_rules! impl_read_methods {
         }
 
         #[allow(unused)]
+        /// Return all certificates involving the user.
+        ///
+        /// ⚠️ This method returns *all* certificates, including the ones that have been
+        /// superseded ! (i.e. you should iter from the last to the first certificate to
+        /// get the current role for a given realm).
```

Comment:
Maybe this logic should be provided as a helper method. I can see that the following logic appears several times:
```rs
realm_roles
            .iter()
            .rev()            
            .find(|role| &role.user_id == cooked.author.user_id())
            .and_then(|role| role.role)
```

Reply:
yes, I wanted to provide a `store.get_realm_last_user_roles()` for this purpose, but that was a bit tricky to correctly implement (it needs to use group by in the query to the sqlite database). So I went the KISS way instead.

I will add an issue about this improvement